src/frontends/tensorflow/src/tf_utils.cpp

```diff
@@ -369,6 +372,115 @@ bool propagate_conditional_flow(const OutputVector& ov_inputs,
+    size_t input_size = ov_inputs.size();
+    // inject condition body graph prior to Loop node
+    // to check condition before to start iterations
+    auto cond_params = cond_model->get_parameters();
+    FRONT_END_GENERAL_CHECK(input_size == cond_params.size(),
+                            "[TensorFlow Frontend] internal error: mismatch number of inputs to While and a number of "
+                            "inputs in a conditional graph");
+    // type setting for body graph parameters is needed for TensorList support since DT_VARIANT type is present
+    // also for more accurate execution_condition variable shape deducing we need shape inference for condition graph
+    for (size_t input_ind = 0; input_ind < input_size; ++input_ind) {
+        cond_params[input_ind]->set_element_type(ov_inputs[input_ind].get_element_type());
+        cond_params[input_ind]->set_partial_shape(ov_inputs[input_ind].get_partial_shape());
+    }
+    cond_model->validate_nodes_and_infer_types();
+
+    auto cond_prior = cond_model->clone();
+    ov::OutputVector ov_outputs;
+    inject_body_model(cond_prior, while_node_name + "/cond", ov_inputs, ov_outputs);
+    FRONT_END_GENERAL_CHECK(
+        ov_outputs.size() == 1,
+        "[TensorFlow Frontend] Internal error or inconsistent model: condition body must contain one Result node.");
+    auto exec_cond = ov_outputs[0];
+    auto trip_count = make_shared<v0::Constant>(element::i32, Shape{}, -1);
+    auto loop = make_shared<v5::Loop>(trip_count, exec_cond);
+
+    // prepare body model to be set for the Loop node
+    // note that condition should be computed on the updated input
+    // because this is while(cond) {} construction,
+    // that is why condition graph is stitched to the body results
+    auto body_params = body_model->get_parameters();
+    auto body_results = body_model->get_results();
+    auto cond_results = cond_model->get_results();
+    auto cond_params_size = cond_params.size();
+    FRONT_END_GENERAL_CHECK(body_params.size() == input_size,
+                            "[TensorFlow Frontend] Internal error or inconsistent model: body graph "
+                            " must have the same number of Parameter nodes as a number of inputs to While.");
+    FRONT_END_GENERAL_CHECK(body_results.size() == input_size,
+                            "[TensorFlow Frontend] Internal error or inconsistent model: body graphs "
+                            " must have the same number of Result nodes as a number of inputs to While.");
+    FRONT_END_GENERAL_CHECK(cond_params.size() == input_size,
+                            "[TensorFlow Frontend] Internal error or inconsistent model: condition graph "
+                            " must have the same number of Parameter nodes as a number of inputs to While.");
+    for (size_t param_ind = 0; param_ind < cond_params_size; ++param_ind) {
+        cond_params[param_ind]->output(0).replace(body_results[param_ind]->input_value(0));
```

Comment:
Just curious, by using 0-index we may expect it could have more than 1 input/output. Do we have any reason to unify this operation by using additional loop for replacing all outputs by inputs? or verify it mandatory has only one input/output?

Reply:
No, we don't need because we replace a value (tensor value) with another one for all consumers of this value.
This is a logic of `replace` method. ov::inputs will be automatically re-connected to the new value.